### ISSUE TITLE
Disallow nesting of shells by default

### ISSUE THREAD

**Issue by shivaraj-bh** [NixOS/nix#10388]
**Is your feature request related to a problem? Please describe.**
Nesting of nix shells results in problems similar to this: https://github.com/NixOS/nix/issues/6140#issuecomment-1986317698. 

**Describe the solution you'd like**
Disallow nesting by default, like `tmux` does with its sessions. Unless there is a strong reason to allow nesting.

**Describe alternatives you've considered**

**Additional context**

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by imadnyc** [comment]
@SuperSandro2000 It seems more that the issue is that installing to profile within a nested shell isn't working correctly on some unprivileged shells. https://github.com/NixOS/nix/issues/6140#issuecomment-1594984535 This comment says it happens on both NixOS and Darwin. I'll try and do more testing later. Even if this is expected, I think that a more descriptive error message about being able to access those files is an improvement.

**Comment by SuperSandro2000** [comment]
What you are describing seems to be a different problem than what the author requested as a new feature in the issue initially.

**Comment by imadnyc** [comment]
I agree, just wanted to mention that it's not Darwin specific. I agree that nested shells is a sane default, and should instead to look at what the original problem is.
